### PR TITLE
fix(helmfile): remove go templating strings on a best effort basis for yaml parsing

### LINKED ISSUE #14344: Helmfile extract fails with local-chart and templated name
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

31.81.0

### Please select which platform you are using if self-hosting.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

14.7.3

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Follow up of https://github.com/renovatebot/renovate/issues/14338#issuecomment-1046904795

When running renovate on a repo with a local Helm chart and `helmfile.yaml` with a templated release name like `name: {{ requiredEnv "RELEASE_NAME" }}`, the `depName` is extracted wrongly (i. e. as `{ '[object Object]': null }`).

See the following for a minimal reproduction:

https://github.com/dominik-bln/renovate-helmfile-dep-name-repro

A simple fix/workaround to make renovate pass from the repository side is quoting the value like so: `name: "{{ requiredEnv "RELEASE_NAME" }}"`

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste any log here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

### PR DESCRIPTION
**Pull request by jandersen-plaid**
## Changes

This PR introduces simple regex parsing (as suggested in the linked Issue) which will remove the Go templating from the helmfile in an attempt to make it closer to pure YAML.

Note that this doesn't actually solve the problem (which is that the go templates need to be parsed), but it does a hacky enough half measure that this should work for most use cases.

## Context

- Closes #14344 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/helmfile/extract.spec.ts [hunk 1]
```diff
@@ -202,5 +202,82 @@ describe('modules/manager/helmfile/extract', () => {
...
+          version: 1.0.0
+          chart: stable/example
+      `;
+      const fileName = 'helmfile.yaml';
+      const result = extractPackageFile(content, fileName, {
+        aliases: {
+          stable: 'https://charts.helm.sh/stable',
+        },
+      });
+      expect(result).toMatchSnapshot({
```
Comment (viceice) [inline lib/modules/manager/helmfile/extract.spec.ts:229]: please use `toMatchObject`

File: lib/modules/manager/helmfile/extract.spec.ts [hunk 2]
```diff
@@ -202,5 +202,82 @@ describe('modules/manager/helmfile/extract', () => {
...
```
Comment (viceice) [inline lib/modules/manager/helmfile/extract.spec.ts:266]: same

File: lib/modules/manager/helmfile/extract.ts [hunk 3]
```diff
@@ -18,7 +18,9 @@ export function extractPackageFile(
   let docs: Doc[];
   const aliases: Record<string, string> = {};
   try {
-    docs = loadAll(content, null, { json: true });
+    docs = loadAll(content.replace(/\s*({{.+}})\s*(\n)/gi, '$2'), null, {
```
Comment (viceice) [inline lib/modules/manager/helmfile/extract.ts:21]: please extract a helper util function, so we can extend later.


```suggestion
    docs = loadAll(content.replace(/^[ \t]*{{.+}}[ \t]*(\n)$/gm, '$1'), null, {
```

your regex is too loose

- `\s` also capture newline
- no need for case insensitive
- use multiline flag, so we can limit to a single line

will create a regex101.com test tomorrow

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/modules/manager/helmfile/extract.spec.ts (hunk 1 above)
Comment (jandersen-plaid) [reply lib/modules/manager/helmfile/extract.spec.ts]: Updated in 38079014c9b0f00d101091228509bb8796df26ad

File: lib/modules/manager/helmfile/extract.ts (hunk 3 above)
Comment (jandersen-plaid) [reply lib/modules/manager/helmfile/extract.ts:21]: I took the liberty of applying your suggestions to a regex 101 [here](https://regex101.com/r/qs8new/1) but I think I may have errored in copying the regex over as it seems to match nothing (or I misconfigured the settings in regex101).

Comment (jandersen-plaid) [reply lib/modules/manager/helmfile/extract.ts:21]: Removing the beginning (`^`) and end of line (`$`) matchers does seem to do the trick though. Keeping them in and removing the newline capture `(\n)` will also work, but that will not replace templates within the line. For example, `version: {{ .Values.example.version }}` would not be changed to `version: `.

File: lib/modules/manager/helmfile/extract.spec.ts (hunk 1 above)
Comment (viceice) [reply lib/modules/manager/helmfile/extract.spec.ts]: Please read https://github.com/renovatebot/renovate/blob/main/.github/contributing.md#resolve-review-comments-instead-of-commenting and reduce noise 🤗

File: lib/modules/manager/helmfile/extract.ts (hunk 3 above)
Comment (viceice) [reply lib/modules/manager/helmfile/extract.ts:21]: What about this?
```suggestion
    docs = loadAll(content.replace(/(^|:)\s*{{.+}}\s*$/gm, ''), null, {
```
https://regex101.com/r/JSiGuZ/1

**Comment by viceice** [comment]
please read https://github.com/renovatebot/renovate/blob/main/.github/contributing.md#resolve-review-comments-instead-of-commenting 🤗

File: lib/modules/manager/helmfile/extract.ts [hunk 4]
```diff
@@ -18,7 +18,9 @@ export function extractPackageFile(
   let docs: Doc[];
   const aliases: Record<string, string> = {};
   try {
-    docs = loadAll(content, null, { json: true }) as Doc[];
+    docs = loadAll(content.replace(/(^|:)\s*{{.+}}\s*$/gm, '$1'), null, {
+      json: true,
+    }) as Doc[];
```
Comment (viceice) [inline lib/modules/manager/helmfile/extract.ts:23]: PLease extract the the regex replace to seperate function.

**Review by viceice — COMMENTED** [review]
Nearly done

**Review by viceice — APPROVED** [review]

**Pull request merged**